lib/bolt/inventory/group.rb

```diff
@@ -315,6 +316,17 @@ def resolve_data_keys(data, target = nil)
           'features' => @plugins.resolve_references(data.fetch('features', [])),
           'plugin_hooks' => @plugins.resolve_references(data.fetch('plugin_hooks', {}))
         }
+
+        # Delete transport config keys that resolved to nil
+        if result['config'].is_a?(Hash)
+          Bolt::Config::Options::TRANSPORT_CONFIG.each_key do |transport|
+            next unless result['config'].key?(transport)
+            transport_config = result['config'][transport]
+            next unless transport_config.is_a?(Hash)
+            result['config'][transport].compact!
+          end
+        end
+
         validate_data_keys(result, target)
```

Comment:
I think we can just move this line further up to validate each value's type before compacting the transport config. Then there's no need for a separate check that `config` is a hash.

Reply:
Updated.